beacon-chain/db/pending_deposits.go

```diff
@@ -113,7 +113,7 @@ func (db *BeaconDB) PrunePendingDeposits(ctx context.Context, b *big.Int) {
 	// added deposit and works backwards.
 	idx := -1
 	for i := depositListSize - 1; i >= 0; i-- {
-		if db.pendingDeposits[i].block.Cmp(b) == -1 {
+		if db.pendingDeposits[i].deposit.MerkleTreeIndex < merkleTreeIndex {
```

Comment:
this relies on the pending deposits being sorted by their indexes, right now they are sorted by the block in which the deposit is logged in

Reply:
Good point!